spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/AnalyzeTableSparkAction.java

```diff
@@ -0,0 +1,159 @@
+
+  private static final Logger LOG = LoggerFactory.getLogger(AnalyzeTableSparkAction.class);
+
+  private final Table table;
+  private Set<String> columns = ImmutableSet.of();
+  private Set<String> types = StandardBlobTypes.blobTypes();
+  private Long snapshotId;
+
+  AnalyzeTableSparkAction(SparkSession spark, Table table) {
+    super(spark);
+    this.table = table;
+  }
+
+  @Override
+  protected AnalyzeTableSparkAction self() {
+    return this;
+  }
+
+  @Override
+  public Result execute() {
+    if (snapshotId == null) {
+      snapshotId = table.currentSnapshot().snapshotId();
+    }
+    String desc = String.format("Analyzing table %s for snapshot id %s", table.name(), snapshotId);
+    JobGroupInfo info = newJobGroupInfo("ANALYZE-TABLE", desc);
+    return withJobGroupInfo(info, this::doExecute);
+  }
+
+  private Result doExecute() {
+    LOG.info("Starting the analysis of {} for snapshot {}", table.name(), snapshotId);
+    List<AnalysisResult> analysisResults =
+        types.stream()
+            .map(
+                statsName -> {
+                  switch (statsName) {
+                    case StandardBlobTypes.APACHE_DATASKETCHES_THETA_V1:
+                      return generateNDVAndCommit();
+                    default:
+                      return ImmutableAnalyzeTable.AnalysisResult.builder()
+                          .type(statsName)
+                          .addAllErrors(Lists.newArrayList("Stats type not supported"))
+                          .build();
+                  }
+                })
+            .collect(Collectors.toList());
+    return ImmutableAnalyzeTable.Result.builder().analysisResults(analysisResults).build();
+  }
+
+  private boolean analyzableTypes(Set<String> columnNames) {
+    return columnNames.stream()
+        .anyMatch(
+            columnName -> {
+              Types.NestedField field = table.schema().findField(columnName);
+              if (field == null) {
+                throw new ValidationException("No column with %s name in the table", columnName);
+              }
+              Type.TypeID type = field.type().typeId();
+              return type == Type.TypeID.INTEGER
+                  || type == Type.TypeID.LONG
+                  || type == Type.TypeID.STRING
+                  || type == Type.TypeID.DOUBLE;
+            });
+  }
+
+  private AnalysisResult generateNDVAndCommit() {
+    try {
+      if (snapshotId == null) {
+        snapshotId = table.currentSnapshot().snapshotId();
+      }
+
+      StatisticsFile statisticsFile =
+          NDVSketchGenerator.generateNDV(
+              spark(), table, snapshotId, columns.toArray(new String[0]));
+      table.updateStatistics().setStatistics(snapshotId, statisticsFile).commit();
+      return ImmutableAnalyzeTable.AnalysisResult.builder()
+          .type(StandardBlobTypes.APACHE_DATASKETCHES_THETA_V1)
+          .build();
+    } catch (IOException ioe) {
+      List<String> errors = Lists.newArrayList();
```

Comment:
Are we only reporting error if we have IOException?  (looks like from writing puffin file).  It seems a bit strange just to catch this specific case, and not other exceptions.

It seems more natural to either catch all exceptions and report error, or else just throw all exceptions, what do you think?